secure_redis/settings.py

```diff
@@ -4,18 +4,34 @@
 from django.core.exceptions import ImproperlyConfigured
 
 
-def get_secure_cache_opts():
-    if hasattr(settings, 'DJANGO_REDIS_SECURE_CACHE_NAME'):
-        cache_name = settings.DJANGO_REDIS_SECURE_CACHE_NAME
+def get_secure_cache_name():
+    """
+    Return the cache name to use for django-rq functionality
+    """
+    if not hasattr(settings, 'DJANGO_REDIS_SECURE_CACHE_NAME'):
+        cache_name = "default"
     else:
-        cache_name = 'default'
+        cache_name = settings.DJANGO_REDIS_SECURE_CACHE_NAME
+    return cache_name
+
 
-    secure_cache_options_settings = settings.CACHES[cache_name].get('OPTIONS')
-    if not secure_cache_options_settings:
+def get_secure_cache_opts():
+    """
+    Return the options dict for the Django cache specified in the
+    `DJANGO_REDIS_SECURE_CACHE_NAME` setting
+    """
+    cache_name = get_secure_cache_name()
+
+    # If `DJANGO_REDIS_SECURE_CACHE_NAME` is explicitly set to `None`, don't
+    # return the options dict, django-rq functionality has been turned off
+    if cache_name is not None:
+        secure_cache_opts = settings.CACHES[cache_name].get('OPTIONS', None)
+        if not secure_cache_opts:
             raise ImproperlyConfigured(
                 'OPTIONS must be defined in settings in secure cache settings!')
-    if secure_cache_options_settings['SERIALIZER'] == 'secure_redis.serializer.SecureSerializer':
-        return secure_cache_options_settings
+
+        if secure_cache_opts.get("SERIALIZER") == 'secure_redis.serializer.SecureSerializer':
```

Comment:
@ptink why do you want to enforce using `secure_redis.serializer.SecureSerializer`?

Reply:
I changed the the dict key lookup back to fail loudly if serializer is missing- otherwise this is functioning the same way it was previously except if cache_name is set to `None` explicitly